src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -367,135 +376,383 @@ public class SqlMutationEngine : IMutationEngine
+                            {
+
+                                hasPerformedUpdate = Convert.ToBoolean(isUpdateResultSetValue);
+                            }
+
+                            // The role with which the REST request is executed can have a database policy defined for the read action.
+                            // In such a case, to get the results back, a select query which honors the database policy is executed.
+                            if (isDatabasePolicyDefinedForReadAction)
+                            {
+                                FindRequestContext findRequestContext = ConstructFindRequestContext(context, upsertOperationResultSetRow, roleName, sqlMetadataProvider);
+                                IQueryEngine queryEngine = _queryEngineFactory.GetQueryEngine(sqlMetadataProvider.GetDatabaseType());
+                                selectOperationResponse = await queryEngine.ExecuteAsync(findRequestContext);
+                            }
+
+                            transactionScope.Complete();
+                        }
+                    }
+
+                    // All the exceptions that can be thrown by .Complete() and .Dispose() methods of transactionScope
+                    // derive from TransactionException. Hence, TransactionException acts as a catch-all.
+                    // When an exception related to Transactions is encountered, the mutation is deemed unsuccessful and
+                    // a DataApiBuilderException is thrown
+                    catch (TransactionException)
                     {
-                        isUpdateResultSet = Convert.ToBoolean(isUpdateResultSetValue);
+                        throw _dabExceptionWithTransactionErrorMessage;
                     }
 
-                    // For MsSql, MySql, if it's not the first result, the upsert resulted in an INSERT operation.
-                    // Even if its first result, postgresql may still be an insert op here, if so, return CreatedResult
-                    if (!isUpdateResultSet ||
-                        (sqlMetadataProvider.GetDatabaseType() is DatabaseType.PostgreSQL &&
-                        PostgresQueryBuilder.IsInsert(resultRow)))
+                    Dictionary<string, object?> resultRow = upsertOperationResultSetRow.Columns;
+
+                    // For all SQL database types, the IS_UPDATE_RESULT_SET field is added to the resultset to indicate if the resultant operation was an update or insert.
+                    // For MsSQL and MySQL, the result set will contain one entry when the resultant operation is update and two entries incase of insert.
+                    // For insert operation, the first result row will not contain any columns. This is used to determine that the resultant operation was an update.
+                    // In this case, IS_UPDATE_RESULT_SET field is added to the result set.
```

Comment:
This comment is confusing to me: 

`// For insert operation, the first result row will not contain any columns. This is used to determine that the resultant operation was an update.
// In this case, IS_UPDATE_RESULT_SET field is added to the result set.`

When you say "in this case", did you mean to refer to the insert operation case, or update?

Reply:
Rephrased the comment. I've removed details about the logic of how `IsUpdateResultSet (IS_UPDATE_RESULT_SET  variable holds IS_UPDATE_RESULT_SET )`  is calculated which I think is not necessary to have in this class. 

Right now, the comment places emphasis on these things
1. The presence of the entry - `<IsUpdateResultSet, true>` in the result set dictionary only in case of update operation. The absence of this entry indicates an insert operation.  
2. For MsSQL, MySQL database types, `IsUpdateResultSet` field is sufficient to determine the type of the resultant operation. However, for PostgreSQL, this entry is always present - both in insert and update. But, PgSQL result sets contain another field (present only in PgSQL and not in MsSQL and MySQL) - `___upsert_op___` that is used to determine the operation type - update/insert.

I think these info should provide the complete context - a) why is an additional field being looked at to determine the resultant operation type b) why is this performed only for PgSQL